changelog.d/6523.feature

```diff
@@ -0,0 +1 @@
+Add option `limit_profile_requests_to_users_who_share_rooms` to prevent requirement of a user sharing a room with another user to query their profile information.
```

Comment:
```suggestion
Add option `limit_profile_requests_to_users_who_share_rooms` to prevent requirement of a local user sharing a room with another user to query their profile information.
```